merge.go

```diff
@@ -72,17 +72,17 @@ type Conditions struct {
 	MaxDollarAmount int64 `json:"maxDollarAmount"`
 }
 
-func MergeFilesWith(files []*File, conditions Conditions) ([]*File, error) {
+func MergeFilesWith(files []File, conditions Conditions) ([]File, error) {
 	return mergeFilesHelper(files, conditions)
 }
 
-func mergeFilesHelper(files []*File, conditions Conditions) ([]*File, error) {
+func mergeFilesHelper(files []File, conditions Conditions) ([]File, error) {
 	fs := &mergableFiles{infiles: files}
 	for i := range fs.infiles {
-		if fs.infiles[i] == nil {
+		if &fs.infiles[i] == nil {
```

Comment:
Looking at these changes again isn't this condition impossible? An address of an object can never be nil.

Reply:
Hm, I think it is possible because that expression evaluates to a pointer(reference) to that object which can be easily compare to nil to check for its nullability.

Reply:
oh yes

you are right